### ISSUE TITLE
Rule proposal: `no-nested-call`

### ISSUE THREAD

**Issue by fisker** [sindresorhus/eslint-plugin-unicorn#1274]
Got the idea from [this line](https://github.com/xojs/xo/blob/7629ce0e887bce9f056eda4300602f90ce2f89bb/index.js#L167).

## Fail

```js
mergeReports(await pMap(await mergeWithFileConfigs(uniq(paths), inputOptions, configFiles), async ({files, options, prettierOptions}) => runEslint(files, buildConfig(options, prettierOptions), {isQuiet: options.quiet})));
```

## Pass

```js
const uniqPaths = uniq(paths);
const mergedConfigs = await mergeWithFileConfigs(
  uniqPaths,
  inputOptions,
  configFiles
);
const reports = await pMap(
  mergedConfigs,
  async ({ files, options, prettierOptions }) =>
    runEslint(
      files, 
      buildConfig(options, prettierOptions),
      {isQuiet: options.quiet}
    )
);

mergeReports(reports);
```

Maybe with a configurable `depth` option.

**Comment by sindresorhus** [comment]
This is now accepted.

**Comment by fisker** [comment]
Let's name it `max-nested-calls` to align with [`max-nested-callbacks` rule](https://eslint.org/docs/rules/max-nested-callbacks)

**Comment by sindresorhus** [comment]
👍
